test/e2e/es_cluster_logging.go

```diff
@@ -0,0 +1,293 @@
+	By("Checking to make sure the Elasticsearch pods are running")
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": "elasticsearch-logging"}))
+	pods, err := c.Pods(ns).List(label)
+	Expect(err).NotTo(HaveOccurred())
+	for _, pod := range pods.Items {
+		err = waitForPodRunning(c, pod.Name)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	// Query against the root URL for Elasticsearch.
+	body, err := c.Get().Prefix("proxy").Resource("services").Name("elasticsearch-logging").DoRaw()
+	Expect(err).NotTo(HaveOccurred())
+	By("Checking to make sure we are talking to an Elasticsearch service.")
+	// Perform a few checks to make sure this looks like an Elasticsearch cluster.
+	var esJSON interface{}
+	if err = json.Unmarshal(body, &esJSON); err != nil {
+		Failf("Failed to unmarshal Elasticsearch response: %v")
+	}
+	esResponse, ok := esJSON.(map[string]interface{})
+	if !ok {
+		Failf("Failed to convert Elasticsearch JSON response to map[string]interface{}")
+	}
+	statusIntf, ok := esResponse["status"]
+	if !ok {
+		Failf("Elasticsearch response has no status field: %v", esResponse)
+	}
+	statusCode, ok := statusIntf.(float64)
+	if !ok {
+		Failf("Expected status to be a float64 but got %T", statusIntf)
+	}
+	if math.Abs(statusCode-200) > 0.001 {
+		Failf("Elasticsearch cluster has a bad status: %v", statusCode)
+	}
+	// Check to see if have a cluser_name field.
+	clusterName, ok := esResponse["cluster_name"]
+	if !ok {
+		Failf("No cluster_name field in Elasticsearch response: %v", esResponse)
+	}
+	if clusterName != "kubernetes_logging" {
+		Failf("Connected to wrong cluster %q (expecting kubernetes_logging)", clusterName)
+	}
+
+	// Now assume we really are talking to an Elasticsearch instance.
+	// Check the cluster health.
+	By("Checking health of Elasticsearch service.")
+	body, err = c.Get().Prefix("proxy").Resource("services").Name("elasticsearch-logging").Suffix("_cluster/health").Param("health", "pretty").DoRaw()
+	Expect(err).NotTo(HaveOccurred())
+	health, err := bodyToJSON(body)
+	Expect(err).NotTo(HaveOccurred())
+	if statusIntf, ok = health["status"]; !ok {
+		Failf("No status field found in cluster health response: %v", health)
+	}
+	status := statusIntf.(string)
+	if status != "green" && status != "yellow" {
+		Failf("Cluster health has bad status: %s", status)
+	}
+
+	// Obtain a list of nodes so we can place one synthetic logger on each node.
+	nodes, err := c.Nodes().List()
+	if err != nil {
+		Failf("Failed to list nodes: %v", err)
+	}
+	nodeCount := len(nodes.Items)
+	if nodeCount == 0 {
+		Failf("Failed to find any nodes")
+	}
+
+	// Create a unqiue root name for the resources in this test to permit
+	// parallel executions of this test.
+	name := "synthlogger-" + string(util.NewUUID())
+	// Replace '-' characters with '_' to prevent the analyzer from breaking apart names.
+	underscoreName := strings.Replace(name, "-", "_", -1)
+
+	podNames := []string{}
```

Comment:
Yes, it is. I've heard the Go gurus advocate both ways (and you can find both in the core libraries), but more often the way I suggested. :) Feel free to ignore if you disagree.